infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesDeployments.java

```diff
@@ -388,19 +392,46 @@ public void onClose(KubernetesClientException cause) {
   }
 
   private void handleStartingPodStatus(CompletableFuture<Void> podRunningFuture, Pod pod) {
-    if (POD_STATUS_PHASE_RUNNING.equals(pod.getStatus().getPhase())) {
-      podRunningFuture.complete(null);
+    PodStatus status = pod.getStatus();
+    String podPhase = status.getPhase();
+    if (POD_STATUS_PHASE_RUNNING.equals(podPhase)) {
+      // check that all the containers are ready...
+      Map<String, String> terminatedContainers = new HashMap<>();
+
+      for (ContainerStatus cs : status.getContainerStatuses()) {
```

Comment:
Does it include the statuses of init containers?

Reply:
Good point - will check that out.

Reply:
I tested, it does not :) -- I was worried the standard init containers would break this.

I think that would be `status.getInitContainerStatuses()` for init containers.

Reply:
Although perhaps a separate issue would be to also track init containers that terminate but don't get status `completed`, in case an init container is failing.